### PR TITLE
Fixes modified clone records breaking

### LINKED ISSUE #9654: Using a genetics data disk in a cloner will break the modified record.
I was trying to mess with everyone's appearances , so I saved data from the cloner to a disk, put it in the genetics console, loaded to a buffer modified, then saved to disk. This had the affect of changing the name of the disk as well.
When I loaded it over a cloning tecord, first it said that the health implant wasn't found anymore, then when I backed out to the records menu, trying to view that record would not open it, the screen would just refresh.
Causes a runtime as well.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #9654

### CODE REVIEW

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -186,11 +186,12 @@
 	H << "<span class='notice'><b>Consciousness slowly creeps over you as your body regenerates.</b><br><i>So this is what cloning feels like?</i></span>"
 
 	H.hardset_dna(ui, se, H.real_name, null, mrace, features)
-	H.faction |= factions
+	if(H)
```
Comment (Aranclanos) [inline code/game/machinery/cloning.dm]: why is this needed?

Comment (MrStonedOne) [inline code/game/machinery/cloning.dm]: I'm assuming hardset_dna sleeps or calls winget or winexists

Other wise this fix is worthless

Comment (AnturK) [inline code/game/machinery/cloning.dm]: In case someone changes the record se value to monkey one Probably better way would be rewrite hardset to return new mob or cloner to handle monkeys properly.

Comment (KazeEspada) [inline code/game/machinery/cloning.dm]: Wouldn't there still be an H even if they are a monkey?

Comment (MrStonedOne) [inline code/game/machinery/cloning.dm]: Nope, it would call monkeyize(), and that deletes the mob.
